pandas/core/strings.py

```diff
@@ -1736,11 +1712,259 @@ def cons_row(x):
                 cons = self._orig._constructor
                 return cons(result, name=name, index=index)
 
-    @copy(str_cat)
-    def cat(self, others=None, sep=None, na_rep=None):
+    def cat(self, others=None, sep=None, na_rep=None, join=None):
+        """
```

Comment:
is there a reason you are now putting this on ``cat`` directly? changing what we are doing with every other function?

Reply:
I don't know where ``str_cat`` is used otherwise, but it's written for arrays. It's cumbersome to add always test if the calling object is a series, that's why I changed ``cat`` directly.

Reply:
I don't think "changing what we are doing with every other function?" is fair at all. I didn't touch ``_get_array_lists`` and ``str_cat`` (except on the initial commits, before the list-of-series case came up), the only function I changed is ``str.cat`` itself, which is what the PR is about.